torch_geometric/explainability/algo/base.py

```diff
@@ -0,0 +1,168 @@
+from abc import abstractmethod
+from typing import Callable, Optional
+
+import torch
+
+from torch_geometric.data import Data
+from torch_geometric.explainability.explanations import Explanation
+
+
+class ExplainerAlgorithm(torch.nn.Module):
+    r"""An abstract class for explainer algorithms."""
+    def __init__(self) -> None:
+        super().__init__()
+
+        self.objective: Callable[
+            [torch.Tensor, torch.Tensor, torch.Tensor, int], torch.Tensor]
+        r"""
+        This method compute the loss to be used for the explanation algorithm.
+
+        This will be set by the Explainer class. It takes the
+        python:function:`loss` and evaluate it at the against the correct
+        reference depending on the explainer mode (model vs phenomenon).
+
+        Args:
+            exp_output (torch.Tensor): the output of the explanation algorithm.
+                (e.g. the forward pass of the model with the mask applied).
+            gnn_output (torch.Tensor): the original output of the GNN.
+            target (torch.Tensor): the target of the GNN.
+            target_index (int): the index of the target to explain.
+        """
+
+    @abstractmethod
+    def loss(self, y_hat: torch.Tensor, y: torch.Tensor) -> torch.Tensor:
+        r"""
+        This method compute the loss to be used for the explanation algorithm.
+        Subclasses should override this method to define their own loss.
+
+        Args:
+            y_hat (torch.Tensor): the output of the explanation algorithm.
+                (e.g. the forward pass of the model with the mask applied).
+            y (torch.Tensor): the target output.
+        """
+
+    @property
+    def accept_new_loss(self) -> bool:
```

Comment:
Not totally sure why we need this function since the loss is defined via implementing the abstract method.